enginetest/enginetests.go

```diff
@@ -1692,13 +1692,59 @@ func TestDropTable(t *testing.T, harness Harness) {
 	_, _, err = e.Query(NewContext(harness), "DROP TABLE not_exist")
 	require.Error(err)
 
+	_, _, err = e.Query(NewContext(harness), "DROP TABLE IF EXISTS not_exist")
+	require.NoError(err)
+
 	t.Run("no database selected", func(t *testing.T) {
```

Comment:
Need another couple sets of tests here:

Current DB selected, drop tables in other db

Drop table in current DB (table name unqualified), and in other DB (table name qualified)